### ISSUE TITLE
[a11y]: ComboBox: clear X icon is not accessible with keyboard

### ISSUE THREAD

**Issue by ntkibm** [carbon-design-system/carbon#19947]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### React version

18.2.0

### Automated testing tool and ruleset

Manual validation

### Description

Expected: To clear a selected item, user should be able to access the clear X icon if they press the Tab key.
Actual: Pressing Tab key takes user out of the ComboBox

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-combobox--default

### Steps to reproduce

- Select an item in ComboBox
- Press Tab key
- Notice it takes you out of ComboBox instead of moving focus onto the clear X icon

Note: I checked in element inspector and it seems the clear X icon has tabindex set to -1.

<img width="1728" height="865" alt="Image" src="https://github.com/user-attachments/assets/8d6ec892-6605-4b19-a4f6-7ab0c4e58d62" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by riddhybansal** [comment]
@ntkibm  Looking into this a bit deeper, this does not appear to be a bug as the ComboBox input is able to be cleared by Escape key. This is listed as a way to clear the combobox in the https://www.w3.org/WAI/ARIA/apg/patterns/combobox/ website

Closing this ticket, but feel free to reopen with any further comments or questions
